src/Eloquent/Model.php

```diff
@@ -259,20 +261,11 @@ public function originalIsEquivalent($key)
             return false;
         }
 
-        if ($this->isDateAttribute($key)) {
-            $attribute = $attribute instanceof UTCDateTime ? $this->asDateTime($attribute) : $attribute;
-            $original = $original instanceof UTCDateTime ? $this->asDateTime($original) : $original;
-
-            return $attribute == $original;
-        }
-
-        if ($this->hasCast($key, static::$primitiveCastTypes)) {
-            return $this->castAttribute($key, $attribute) ===
-                $this->castAttribute($key, $original);
+        try {
+            return fromPHP([$attribute]) === fromPHP([$original]);
+        } catch (UnexpectedValueException $e) {
```

Comment:
Do you have cases of `fromPHP` throwing a `MongoDB\Driver\Exception\UnexpectedValueException`? If the attribute can't be converted to BSON, I suspect the `update` operation will fail.

Reply:
> If the attribute can't be converted to BSON, I suspect the update operation will fail.

What is "update operation" referring to? I'm missing the context from this PR.

Looking through [`php_phongo_zval_to_bson_internal()`](https://github.com/mongodb/mongo-php-driver/blob/1.16.2/src/phongo_bson_encode.c#L402) there are some cases that come to mind:

 * Attempting to encode prohibited values as root-level documents (e.g. enum, types)
 * Keys contain null bytes
 * Unexpected return value from a `bsonSerialize()` method (if not caught by PHP itself)
 * Encountering an unexpected BSON type instance
 * Encoding a non-backed enum
 * Encoding a string that fails UTF-8 validation
 * Encountering recursion during encoding

None of these should happen under normal operation (there's a joke about "unexpected" there somewhere).